presto-product-tests/src/main/java/com/facebook/presto/tests/convention/TestShowPartitions.java

```diff
@@ -0,0 +1,85 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.convention;
+
+import com.google.inject.Inject;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.Requirement;
+import com.teradata.tempto.RequirementsProvider;
+import com.teradata.tempto.configuration.Configuration;
+import com.teradata.tempto.fulfillment.table.MutableTableRequirement;
+import com.teradata.tempto.fulfillment.table.MutableTablesState;
+import com.teradata.tempto.fulfillment.table.TableDefinition;
+import com.teradata.tempto.fulfillment.table.hive.HiveDataSource;
+import com.teradata.tempto.fulfillment.table.hive.HiveTableDefinition;
+import com.teradata.tempto.query.QueryResult;
+import org.testng.annotations.Test;
+
+import java.sql.SQLException;
+
+import static com.facebook.presto.tests.TestGroups.BASIC_SQL;
+import static com.teradata.tempto.Requirements.compose;
+import static com.teradata.tempto.assertions.QueryAssert.Row.row;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.fulfillment.table.TableRequirements.mutableTable;
+import static com.teradata.tempto.fulfillment.table.hive.InlineDataSource.createResourceDataSource;
+import static com.teradata.tempto.fulfillment.table.hive.InlineDataSource.createStringDataSource;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.System.currentTimeMillis;
+
+public class TestShowPartitions
+        extends ProductTest
+        implements RequirementsProvider
+
+{
+    @Inject
+    private MutableTablesState tablesState;
+
+    private static final String TABLE_NAME = "partitioned_table";
+
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return compose(mutableTable(generateTableDefinition(), TABLE_NAME, MutableTableRequirement.State.CREATED));
+    }
+
+    private static TableDefinition generateTableDefinition()
+    {
+        StringBuilder sb = new StringBuilder();
+        sb.append("CREATE EXTERNAL TABLE %NAME%(");
+        sb.append("   col INT");
+        sb.append(") ");
+        sb.append("PARTITIONED BY (part_col INT) ");
+        sb.append(" STORED AS ORC");
+
+        HiveDataSource dataSource = createResourceDataSource(TABLE_NAME, "" + currentTimeMillis(), "com/facebook/presto/tests/hive/data/single_int_column/data.orc");
+        HiveDataSource invalidData = createStringDataSource(TABLE_NAME, "" + currentTimeMillis(), "INVALID DATA");
```

Comment:
I looked through the product tests, and it seems that this style is used in only one other place.  I am concerned anytime I see wall clock milliseconds used for entropy since there are so few bits of entropy in wall clock milliseconds.